osv2/executor.go

```diff
@@ -26,45 +26,52 @@ func NewExecutor(client *opensearch.Client) *Executor {
 // will be returned.
 // An error can be returned if:
 //   - The request to OpenSearch fails
-//   - The results json cannot be unmarshalled
-func (e *Executor) MGet(ctx context.Context, req *opensearchtools.MGetRequest) (*opensearchtools.OpenSearchResponse[opensearchtools.MGetResponse], error) {
+//   - The results JSON cannot be unmarshalled
+func (e *Executor) MGet(ctx context.Context, req *opensearchtools.MGetRequest) (resp opensearchtools.OpenSearchResponse[opensearchtools.MGetResponse], err error) {
```

Comment:
Are you just naming the return variable so you don't have to return nil? I'm not sure I follow that value of this change.

Reply:
It has to do with trying to keep things in the stack rather than the heap which can be more optimal for performance.